casr/src/bin/casr-ubsan.rs

```diff
@@ -0,0 +1,456 @@
+                .value_parser(clap::value_parser!(u64).range(1..))
+        )
+        .arg(
+            Arg::new("input")
+                .short('i')
+                .long("input")
+                .action(ArgAction::Set)
+                .required(true)
+                .num_args(1..)
+                .value_name("INPUT_DIRS")
+                .help("Target input directory list")
+                .value_parser(move |arg: &str| {
+                    let i_dir = Path::new(arg);
+                    if !i_dir.exists() {
+                        let mut err = clap::Error::new(ErrorKind::ValueValidation);
+                        err.insert(ContextKind::InvalidValue, ContextValue::String("Input directory doesn't exist.".to_owned()));
+                        return Err(err);
+                    }
+                    if !i_dir.is_dir() {
+                        let mut err = clap::Error::new(ErrorKind::ValueValidation);
+                        err.insert(ContextKind::InvalidValue, ContextValue::String("Input path should be a directory.".to_owned()));
+                        return Err(err);
+                    }
+                    Ok(i_dir.to_path_buf())
+                })
+        )
+        .arg(
+            Arg::new("output")
+                .short('o')
+                .long("output")
+                .action(ArgAction::Set)
+                .required(true)
+                .value_name("OUTPUT_DIR")
+                .value_parser(clap::value_parser!(PathBuf))
+                .help("Output directory with triaged reports")
+        )
+        .arg(
+            Arg::new("ARGS")
+                .action(ArgAction::Set)
+                .required(false)
+                .num_args(1..)
+                .last(true)
+                .help("Add \"-- <path> <arguments>\" to run"),
+        )
+        .get_matches();
+
+    // Init log.
+    util::initialize_logging(&matches);
+
+    debug!("Get args");
+    // Get input dir list
+    let input_dirs: Vec<_> = matches.get_many::<PathBuf>("input").unwrap().collect();
+    // Get output dir
+    let output_dir = matches.get_one::<PathBuf>("output").unwrap();
+    if !output_dir.exists() {
+        fs::create_dir_all(output_dir).with_context(|| {
+            format!("Couldn't create output directory {}", output_dir.display())
+        })?;
+    } else if !output_dir.is_dir() {
+        bail!("Output directory must be a directory");
+    } else if output_dir.read_dir()?.next().is_some() {
+        bail!("Output directory is not empty.");
+    }
+    // Get program args.
+    let argv: Vec<&str> = if let Some(argvs) = matches.get_many::<String>("ARGS") {
+        argvs.map(|s| s.as_str()).collect()
+    } else {
+        bail!("Wrong arguments for starting program");
+    };
+
+    // Get timeout
+    let timeout = if let Some(timeout) = matches.get_one::<u64>("timeout") {
+        *timeout
+    } else {
+        0
+    };
+
+    // Get input path list
+    debug!("Get input path list");
```

Comment:
Remove. Does not provide any useful information.

Reply:
Done